boilerplate/app/utils/crash-reporting.ts

```diff
@@ -0,0 +1,69 @@
+/**
+ * If you're using Sentry
+ *   RN   https://docs.sentry.io/platforms/react-native/
+ *   Expo https://docs.expo.dev/guides/using-sentry/
+ */
+// import * as Sentry from "sentry-expo"
+// import * as Sentry from "@sentry/react-native"
+
+/**
+ * If you're using Crashlytics: https://rnfirebase.io/crashlytics/usage
+ */
+// import crashlytics from "@react-native-firebase/crashlytics"
+
+/**
+ * If you're using Bugsnag:
+ *   RN   https://docs.bugsnag.com/platforms/react-native/)
+ *   Expo https://docs.bugsnag.com/platforms/react-native/expo/
+ */
+// import Bugsnag from "@bugsnag/react-native"
+// import Bugsnag from "@bugsnag/expo"
+
+/**
+ *  This is where you put your crash reporting service initialization code to call in `./app/app.tsx`
+ */
+export const initCrashReporting = () => {
+  // Sentry.init({
+  //   dsn: "YOUR DSN HERE",
+  //   enableInExpoDevelopment: true,
+  //   debug: true, // If `true`, Sentry will try to print out useful debugging information if something goes wrong with sending the event. Set it to `false` in production
+  // })
+  // Bugsnag.start("YOUR API KEY")
+}
+
+/**
+ * Error classifications used to sort errors on error reporting services.
+ */
+export enum ErrorType {
+  /**
+   * An error that would normally cause a red screen in dev
+   * and force the user to sign out and restart.
+   */
+  FATAL = "Fatal",
+  /**
+   * An error caught by try/catch where defined using Reactotron.tron.error.
+   */
+  HANDLED = "Handled",
+}
+
+/**
+ * Manually report a handled error.
+ */
+export const reportCrash = (error: any, type: ErrorType = ErrorType.FATAL) => {
+  const message = error.message || "Unknown"
+  console.error(error)
+  console.log(message, type)
+
+  __DEV__ && console.tron.log(error)
```

Comment:
Minor nit -- if you use the && format, Metro won't strip it out of the final build (although it will still work as expected). If you change it to an if statement, Metro will strip it out of the prod build.

```suggestion
  if (__DEV__) {
    console.tron.log(error)
  }
```

Reply:
We should probably also early exit here. We don't want to report bugs in dev.